plugins/inputs/opcua/opcua.go

```diff
@@ -1,576 +0,0 @@
-//go:generate ../../../tools/readme_config_includer/generator
```

Comment:
This PR is quite large and hard to review. For example, This file was both moved and modified so changes don't show up in diffs and it forces reviewers to find where code was moved to manually and also check how it was changed manually.

I really appreciate all the work you put into the PR and I'm trying to think of a way to make reviewing easier without making too much more work for you. What do you think about splitting this PR into two or more PRs that make smaller steps so diffs are more useful?

If splitting into multiple PRs doesn't sound good, what about rewriting the branch history (git rebase -i) and squashing commits into high level steps (factor out code from inputs.opcua into common, fix bug x, fix docs typos, add listener plugin, etc) so it's possible to review each commit in sequence? It's not practical to review each commit now because there are so many commits in the branch.

Thanks!

Reply:
A big thank you for putting this together as well. I agree with Dave that it would be helpful if you could split this up to aid our review and ensure aren't introducing any hard to find regressions. Thanks in advanced!

Reply:
Yes, I understand. It is a very big PR. The process of refactoring was a bit messy as well, as I was not at all familiar yet with the Telegraf code base. I will see what I can do with rebasing interactively.

Reply:
@reimda @powersj I have reworked the commit history. I think this will make it a lot easier.